workspace/2019 DeepSpace/src/main/java/frc/robot/Robot.java

```diff
@@ -115,103 +184,48 @@ public void teleopPeriodic() {
-                    break;
-                case 3:
-                    m_lifter.GoToThirdBallLevel();
-                    System.out.println("Ball Third Level");
-                    break;
-            }
+    }
+    
+    @Override
+    public void teleopInit() {
+        int macroIndex = 0;
+        m_compressor.start() // If it failed in robotInit(), just in case.
+    }
+                
+    @Override
+    public void teleopPeriodic() {
+        m_robotDrive.driveCartesian(kMotorPowerLevel * m_stick.getX(),
+                                    kMotorPowerLevel * m_stick.getY(),
+                                    kMotorPowerLevel * m_stick.getZ(), 0.0);
+        if (m_stick.getRawButtonPressed(kButtonA)) {
+            m_grabber.Grab();
         }
-        if (m_stick.getPOV() == 0) {
-            if (lifterLevelHatch < 7 && !currentlyPressed && ballMode != true){
-                if(lifterLevelHatch != 3 ) {
-                    lifterLevelHatch++;
-                    currentlyPressed = true;}
-            } 
+        if (m_stick.getRawButtonPressed(kButtonB)) {
+            m_grabber.Eject();
         }
-        else if (m_stick.getPOV() == 180) {
-            if (lifterLevelHatch > 0 && !currentlyPressed && ballMode != true){
-                if(lifterLevelHatch != 0) {
-                    lifterLevelHatch--;
-                    currentlyPressed = true;}
-            } 
+        if (m_stick.getRawButtonPressed(kButtonX)) {
+            m_sucker.Suck(true);
         }
-        else {
-           currentlyPressed = false;
+        if (m_stick.getRawButtonPressed(kButtonY)) {
+            m_sucker.Suck(false);
         }
-        if(ballMode != true) {
-            switch(lifterLevelHatch) {
-                case 0:
-                    m_lifter.GoToBottom();
-                    System.out.println("Bottom");
-                    break;
-                case 1:
-                    m_lifter.GoToFirstHatchLevel();
-                    System.out.println("Hatch First Level");
-                    break;
-                case 2:
-                    m_lifter.GoToSecondHatchLevel();
-                    System.out.println("Hatch Second Level");
-                    break;
-                case 3:
-                    m_lifter.GoToThirdHatchLevel();
-                    System.out.println("Hatch Third Level");
-                    break;
+        if (m_stick.getRawButtonPressed(kButtonstart)) { // This code doesn't exist in the auto stuff, for good reason
+            macroIndex++;  // I've had to edit this code, because the Climb() function is idealized
+            if (macroIndex == 1)
+                m_walker.RaiseRobot();
+            if (macroIndex == 2)
+                m_walker.Walk();
+            if (macroIndex == 3) //By idealized, I mean, the robot has to drive during this step.
+                m_walker.RetractBackLegs(); // Climb() doesn't give the operator time to do so.
```

Comment:
The steps for climbing are:
1- drive backwards to platform
2- raise robot
3- "walk", where the back legs are tilted so that the robot falls backwards onto the platform the robot should fall back enough such that the wheels hit the platform
4- retract back legs, in preparation for driving backwards
5- drive backwards until robot is fully above platform
6- retract front legs. front wheels of robot will fall and rest on the platform edge

the Climb() function basically went from one step to the next without waiting in between. It wouldn't have even finished waiting for the function of extension/retraction to execute. If you look at the 2018 code, you'll see a macro timer, which was necessary because each step took time to complete. Separating the Climb() function into its steps allows the operator to climb in his own timing.

Reply:
This is discussed in issue #24.
https://github.com/garykac/amhs-robotics-4681/blob/2aa4fd3c6ba7218b489a7fc224bba6bec184faae/workspace/2018-FIRST_Power_Up/2018RampBot/src/org/usfirst/frc/team4681/robot/Robot.java#L331-L335
https://github.com/garykac/amhs-robotics-4681/blob/2aa4fd3c6ba7218b489a7fc224bba6bec184faae/workspace/2018-FIRST_Power_Up/2018RampBot/src/org/usfirst/frc/team4681/robot/Robot.java#L348-L382

This is the relevant 2018 robot code that demonstrates why we can't have a single climb function. Each step needs to be carried out separately.